include/zephyr/drivers/clock_control/it8xxx2_clock_control.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2022 ITE Corporation. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DRIVERS_CLOCK_CONTROL_IT8XXX2_CLOCK_CONTROL_H_
+#define ZEPHYR_INCLUDE_DRIVERS_CLOCK_CONTROL_IT8XXX2_CLOCK_CONTROL_H_
+
+#include <zephyr/types.h>
+#include "chip_chipregs.h"
+
+#define SWUC_CLOCK_GATING BIT(4)
+
+#define CGCTRL3R_ALWAYS_1 BIT(6)
+#define PECI_CLOCK_GATING BIT(3)
+#define UART_CLOCK_GATING BIT(2)
+#define SSPI_CLOCK_GATING BIT(1)
+#define DBGR_CLOCK_GATING BIT(0)
+#define PECI_SSPI_DBGR_CLOCK_GATING \
+	(PECI_CLOCK_GATING | SSPI_CLOCK_GATING | \
+	DBGR_CLOCK_GATING | CGCTRL3R_ALWAYS_1)
+
+#define SMB_CHF_CLOCK_GATING BIT(7)
+#define SMB_CHE_CLOCK_GATING BIT(6)
+#define SMB_CHD_CLOCK_GATING BIT(5)
+#define SMB_CHC_CLOCK_GATING BIT(4)
+#define SMB_CHB_CLOCK_GATING BIT(3)
+#define SMB_CHA_CLOCK_GATING BIT(2)
+#define SMB_ALL_CHANNELS_CLOCK_GATING \
+	(SMB_CHA_CLOCK_GATING | SMB_CHB_CLOCK_GATING | SMB_CHC_CLOCK_GATING | \
+	SMB_CHD_CLOCK_GATING | SMB_CHE_CLOCK_GATING | SMB_CHF_CLOCK_GATING)
+
+#define SPI_SLAVE_CLOCK_GATING BIT(3)
+#define PD1_CLOCK_GATING BIT(1)
+#define PD0_CLOCK_GATING BIT(0)
+#define PD_ALL_PORTS_CLOCK_GATING (PD0_CLOCK_GATING | PD1_CLOCK_GATING)
+
+#define JTAG_CHB_CLOCK_GATING BIT(3)
+#define FPU_CHA_CLOCK_GATING BIT(2)
+
+#ifndef _ASMLANGUAGE
+struct it8xxx2_clock_control_cells {
+	uint8_t offset;
+	uint8_t bitmask;
+	uint8_t always;
+};
+
+#endif
+
+
+/* Clock Control Sybsystem Options */
+enum it8xxx2_clock {
+	IT8XXX2_NULL,
+	IT8XXX2_PLL,
+	IT8XXX2_CPU,
+	IT8XXX2_SMB,
+	IT8XXX2_CLOCK_NUM_MAX = IT8XXX2_SMB,
+};
+
+struct it8xxx2_clkctrl_subsys_t {
+	uint8_t clk_opt;
+};
+
+struct it8xxx2_clkctrl_subsys_rate_t {
+	uint32_t clk_rate;
+};
+
+struct it8xxx2_clkctrl_freq_t {
+	uint8_t divisor;
+	uint32_t frequency;
+};
+
+#define ECPM_IT8XXX2_REG_BASE \
+((struct ecpm_it8xxx2_regs *)DT_REG_ADDR(DT_NODELABEL(ecpm)))
+
+void chip_pll_ctrl(enum chip_pll_mode mode);
```

Comment:
is everything meant to be public API here? If so please document things properly.

Reply:
My bad, they shouldn't be here, these bits definitions should be somewhere else.
Thanks for correction, sir.